mmv1/templates/terraform/constants/resource_dns_resource_record_set.go.erb

```diff
@@ -1,19 +1,50 @@
 func rrdatasDnsDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
-	if d.Get("type") == "AAAA" {
-		return ipv6AddressforDnsDiffSuppress(k, old, new, d)
+	oo, nn := d.GetChange("rrdatas")
+	if oo == nil || nn == nil {
+		return false
 	}
-	if d.Get("type") == "MX" {
-		return (strings.ToLower(old) == strings.ToLower(new))
-	}
-	if d.Get("type") == "TXT" {
-		return strings.ToLower(strings.Trim(old, `"`)) == strings.ToLower(strings.Trim(new, `"`))
+
+	olist := convertStringArr(oo.([]interface{}))
+	nlist := convertStringArr(nn.([]interface{}))
+
+	parsefunc := func(x string) string {
+		switch d.Get("type") {
+		case "AAAA":
+			// parse ipv6 to a key from one list
+			return net.ParseIP(x).String()
+		case "MX":
+			return strings.ToLower(x)
+		case "TXT":
+			return strings.ToLower(strings.Trim(x, `"`))
+		default:
+			return x
+		}
 	}
-	return false
+	return rrdatasforDnsDiffSuppress(olist, nlist, parsefunc, d)
 }
 
-func ipv6AddressforDnsDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
-	oldIp := net.ParseIP(old)
-	newIp := net.ParseIP(new)
+type parseStr func(string) string
```

Comment:
Do we need a new type for this? I think we can just have `parseFunc func(string) string`, as a parameter to the method?

Like: https://play.golang.org/p/VIh6jhUfT3w